### PR TITLE
En 4877 data field as byte array

### PR DESCRIPTION
**Pull request by bogdan-rosianu**

### CODE REVIEW

File: core/genesis/genesisBlockCreator.go [hunk 1]
```diff
@@ -272,7 +272,7 @@ func deploySystemSmartContracts(
 		RcvAddr:   make([]byte, addrConv.AddressLen()),
 		GasPrice:  0,
 		GasLimit:  0,
-		Data:      hex.EncodeToString([]byte("deploy")) + "@" + hex.EncodeToString(factory.SystemVirtualMachine),
+		Data:      []byte(hex.EncodeToString([]byte("deploy")) + "@" + hex.EncodeToString(factory.SystemVirtualMachine)),
```
Comment (raduchis) [inline core/genesis/genesisBlockCreator.go]: is it ok that also "deploy" is hexEncoded? The next method "stake" is not hex.Encoded just the params

File: data/transaction/transaction_test.go [hunk 2]
```diff
@@ -68,15 +68,15 @@ func TestTransaction_GetValue(t *testing.T) {
...
-func TestTransaction_SetData(t *testing.T) {
-	t.Parallel()
-
-	data := "data"
-	tx := &transaction.Transaction{}
-	tx.SetData(data)
-
-	assert.Equal(t, data, tx.Data)
-}
+//func TestTransaction_SetData(t *testing.T) {
```
Comment (raduchis) [inline data/transaction/transaction_test.go]: Why was this test commented?

File: process/block/shardblock_test.go [hunk 3]
```diff
@@ -2867,13 +2867,13 @@ func TestShardProcessor_CreateMiniBlocksShouldWorkWithIntraShardTxs(t *testing.T
 	txProcessorMock := &mock.TxProcessorMock{
 		ProcessTransactionCalled: func(transaction *transaction.Transaction, round uint64) error {
 			//execution, in this context, means moving the tx nonce to itx corresponding execution result variable
-			if transaction.Data == string(txHash1) {
+			if bytes.Equal(transaction.Data, txHash1) {
```
Comment (raduchis) [inline process/block/shardblock_test.go]: +1 for the bytes.equal, not string(transaction.Data)

File: data/transaction/transaction_test.go (hunk 2 above)
Comment (bogdan-rosianu) [reply data/transaction/transaction_test.go]: this was commented before I rebuilt the capn schema and I forgot to uncomment it.

File: core/genesis/genesisBlockCreator.go (hunk 1 above)
Comment (bogdan-rosianu) [reply core/genesis/genesisBlockCreator.go]: I am not sure about this. I just cast the resulting string to a byte array. Maybe ask @sasurobert

File: api/transaction/routes.go [hunk 4]
```diff
@@ -13,8 +13,8 @@ import (
 
 // TxService interface defines methods that can be used from `elrondFacade` context variable
 type TxService interface {
-	CreateTransaction(nonce uint64, value string, receiverHex string, senderHex string, gasPrice uint64, gasLimit uint64, data string, signatureHex string, challenge string) (*transaction.Transaction, error)
-	SendTransaction(nonce uint64, sender string, receiver string, value string, gasPrice uint64, gasLimit uint64, code string, signature []byte) (string, error)
+	CreateTransaction(nonce uint64, value string, receiverHex string, senderHex string, gasPrice uint64, gasLimit uint64, data []byte, signatureHex string, challenge string) (*transaction.Transaction, error)
+	SendTransaction(nonce uint64, sender string, receiver string, value string, gasPrice uint64, gasLimit uint64, txData []byte, signature []byte) (string, error)
```
Comment (SebastianMarian) [inline api/transaction/routes.go]: data instead txdata?

File: facade/elrondNodeFacade.go [hunk 5]
```diff
@@ -163,12 +163,12 @@ func (ef *ElrondNodeFacade) CreateTransaction(
 	senderHex string,
 	gasPrice uint64,
 	gasLimit uint64,
-	data string,
+	txData []byte,
```
Comment (SebastianMarian) [inline facade/elrondNodeFacade.go]: data instead txdata?

File: facade/elrondNodeFacade.go [hunk 6]
```diff
@@ -179,11 +179,11 @@ func (ef *ElrondNodeFacade) SendTransaction(
 	value string,
 	gasPrice uint64,
 	gasLimit uint64,
-	transactionData string,
+	txData []byte,
```
Comment (SebastianMarian) [inline facade/elrondNodeFacade.go]: data instead txdata?

File: facade/elrondNodeFacade_test.go [hunk 7]
```diff
@@ -224,12 +224,12 @@ func TestElrondNodeFacade_SetSyncer(t *testing.T) {
 func TestElrondNodeFacade_SendTransaction(t *testing.T) {
 	called := 0
 	node := &mock.NodeMock{}
-	node.SendTransactionHandler = func(nonce uint64, sender string, receiver string, amount string, code string, signature []byte) (string, error) {
+	node.SendTransactionHandler = func(nonce uint64, sender string, receiver string, amount string, txData []byte, signature []byte) (string, error) {
```
Comment (SebastianMarian) [inline facade/elrondNodeFacade_test.go]: data instead txdata?

File: data/receipt/capnp/schema.capnp.go [hunk 8]
```diff
@@ -6,9 +6,8 @@ import (
 	"bufio"
 	"bytes"
 	"encoding/json"
-	"io"
-
 	C "github.com/glycerine/go-capnproto"
+	"io"
```
Comment (raduchis) [inline data/receipt/capnp/schema.capnp.go]: fix imports, maybe

**Pull request merged**